### ISSUE TITLE
Loss function and method for importance

### ISSUE THREAD

**Issue by lionelkusch** [mind-inria/hidimstat#49]
For LOCO, CPI and PI, the classes have a loss function as a parameter to evaluate the importance of the variables.
The various loss functions must be used with a specific estimator “method” (“predict”, “predict_proba”, “decision_function” and “transform”).

For the time being, the user needs to know which loss functions are associated with which “method”.

We have two options:
1. Stay as we are and add documentation to indicate which method is to be used with which loss functions. 
2. Remove the possibility for users to choose any loss function and provide a list of loss functions with their associated methods. 

This question remains open.

**Comment by lionelkusch** [comment]
This question will evolve with the integration of p-value methods #48 .
The definition of p-value is connected to the definition of the loss function and the estimator "method".

**Comment by bthirion** [comment]
I would accept all estimators in sklearn ; (Do they need to have a `predict_proba` method ?)
And document (and test ! ) choices that are suitable.
People may want to use other estimators, but it is at their own risk.

**Comment by jpaillard** [comment]
As mentioned in #152, we should have a test to ensure that the loss provided is 1. a valid loss function in general and 2. a valid loss function for the method considered.

**Comment by GrituX** [comment]
The current system allows for user-made custom loss functions, which can be experimental. In that regard, it may be hard to check if the provided loss is valid. Do we really need to make that check ?

**Comment by bthirion** [comment]
@Remi-Gau I'd be happy to hear your opinion on this.
